src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/ValidationErrors.kt

```diff
@@ -6,3 +6,21 @@ value class ValidationErrors(private val errorMap: MutableMap<String, String>) :
 }
 
 fun singleValidationErrorOf(propertyNameToMessage: Pair<String, String>) = ValidationErrors().apply { this[propertyNameToMessage.first] = propertyNameToMessage.second }
+
+class ValidatedScope<EntityType> {
+  private val validationErrors = ValidationErrors()
+
+  val validationErrorsPresent: Boolean
+    get() = validationErrors.any()
+
+  val accumulatedFieldValidationError: ValidatableActionResult.FieldValidationError<EntityType> = ValidatableActionResult.FieldValidationError(validationErrors)
+
+  infix fun success(entity: EntityType) = ValidatableActionResult.Success(entity)
+  infix fun generalError(message: String) = ValidatableActionResult.GeneralValidationError<EntityType>(message)
+  infix fun String.hasValidationError(message: String) = validationErrors.put(this, message)
+  infix fun String.hasSingleValidationError(message: String) = ValidatableActionResult.FieldValidationError<EntityType>(singleValidationErrorOf(this to message))
```

Comment:
Do we even need `this singleValidationErrorOf` now? As we're initializing an empty `validationErrors` in the scope?

Reply:
It's still used in this file for `hasSingleValidationError`, it can't be succintly inlined though because of the apply call vs the receiver function in the infix both being a `this` so think it's clearest to keep it like this.  Will make it `private` though.